pkg/stanza/operator/input/file/config.go

```diff
@@ -82,8 +84,8 @@ func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
 		toBody:         toBody,
 		preEmitOptions: preEmitOptions,
 	}
-
-	input.fileConsumer, err = c.Config.Build(logger, input.emit)
+	input.fileConsumer, err = c.Config.Build(logger, input.emit,
+		fileconsumer.WithMultilineSplitter(c.EncodingConfig, c.Flusher, c.MultilineConfig))
```

Comment:
`c.EncodingConfig` and `c.Flusher` are fields that already belong to the fileconsumer.Config. It seems like we should be able to ask the config for a splitter based only on the one parameter that it does not have.

Reply:
> @atingchen, thanks for doing this. I think this looks a lot like what I was expected, but I'm having a hard time feeling confident in the changes. This is in part because this areas of the codebase is a bit tangled up.
> 
> We may have to iterate on this one slowly and carefully. Another thing that may help, if you have ideas, is to make smaller changes that make sense on their own and work towards this design.

Thank you for your CR. I will try to split it into several small PRs to gradually implement this feature.